src/services/utilServices/SitesService.js

```diff
@@ -29,12 +29,38 @@ const ISOMER_ADMIN_REPOS = [
 ]
 
 class SitesService {
-  constructor({ gitHubService, configYmlService }) {
+  constructor({
+    gitHubService,
+    configYmlService,
+    usersService,
+    isomerAdminsService,
+  }) {
     this.githubService = gitHubService
     this.configYmlService = configYmlService
+    this.usersService = usersService
+    this.isomerAdminsService = isomerAdminsService
+  }
+
+  async getEmailUserSites(userId) {
+    const user = await this.usersService.findSitesByUserId(userId)
+    if (user) {
+      const { site_members: siteMemberEntries } = user
+      return siteMemberEntries.map((entry) => {
+        const repoData = entry.repo
+        return repoData.name
+      })
+    }
+    return []
   }
 
   async getSites(sessionData) {
+    const isEmailUser = sessionData.isEmailUser()
+    let retrievedSitesByEmail = []
+    const { isomerUserId: userId } = sessionData
+    const isAdminUser = !!(await this.isomerAdminsService.getByUserId(userId))
+    if (!isAdminUser && isEmailUser)
+      retrievedSitesByEmail = await this.getEmailUserSites(userId)
+
```

Comment:
not sure if this is out of scope for this PR but wdyt about refactoring this to 2 methods called `getSitesForUser` and `getSitesForAdmin`? 

this method is p confusing to read through at the moment, especially with the nested `map/filter` operations ._.

Reply:
Same as above, I think this part might have been confusing, so I refactored the code - isAdminUser is supposed to simply allow the result of the github api query to return all sites. Changed in https://github.com/isomerpages/isomercms-backend/pull/480/commits/86cdc315049dfd6dead583c715fefe02329822ce, lmk if it's still confusing

Reply:
double checking here - the reason why `!isEmailUser` is part of the condition (`isAdminuser || !isEmailUser`) is because we use the user's own github access token to retrieve the sites and github will resolve permissions for us on their end?